### ISSUE TITLE
False negative ParadoxicalCondition using function call in switch()

### ISSUE THREAD

**Issue by ygottschalk** [vimeo/psalm#3808]
Feature request to detect ParadoxicalConditions in a switch using a call.

function-call: https://psalm.dev/r/1bdb320ffa
method-call: https://psalm.dev/r/8144392cca
static-method-call: https://psalm.dev/r/b9596de50b
closure call: https://psalm.dev/r/262b9cc572

Simple workaround:
```
$returnVal = call();
switch($returnVal) { [...] }
```

**Comment by muglug** [comment]
ha ok, this currently breaks down because Psalm converts switch calls to roughly-equivalent if statements:

```php
if (foo() == 1) {
} elseif (foo() == 2)) {
} elseif (foo() == 3) {
} else {}
```

which obviously doesn't have the same guarantees.

The solution would be to swap the function call for a `$__tmp_switch_var_<file_offset>` variable fetch, allowing proper reconciliation of the result.

**Comment by ygottschalk** [comment]
https://psalm.dev/r/754a88c984

I marked the function as pure... shouldn't that give those guarantees and therefore make psalm report the paradox?
